cli/src/main/java/com/michelin/ns4kafka/cli/services/ResourceService.java

```diff
@@ -71,14 +76,25 @@ public Resource getSingleResourceWithType(ApiResource apiResource, String namesp
 
     public HttpResponse<Resource> apply(ApiResource apiResource, String namespace, Resource resource, boolean dryRun) {
         try {
+            // If the resource is a subject, load the schema
+            if (resource.getKind().equals("Schema")) {
+                fileService.loadSchemaFileContent(resource);
+            }
+
             if (apiResource.isNamespaced()) {
                 return namespacedClient.apply(namespace, apiResource.getPath(), loginService.getAuthorization(), resource, dryRun);
             } else {
                 return nonNamespacedClient.apply(loginService.getAuthorization(), apiResource.getPath(), resource, dryRun);
             }
         } catch (HttpClientResponseException e) {
             formatService.displayError(e, apiResource.getKind(), resource.getMetadata().getName());
+        } catch (NoSuchFileException e) {
+            System.out.printf("Failed : %s/%s %s%n", apiResource.getKind(), resource.getMetadata().getName(),
+                    "Cannot find schema. Schema path must be relative to the CLI, not to the resource file");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
         }
```

Comment:
Throw an exception with your custom message in fileService.loadSchemaFileContent and use ``formatService.displayError`` instead:

```suggestion
        } catch (HttpClientResponseException | YourCustomFileLoadingException e) {
            formatService.displayError(e, apiResource.getKind(), resource.getMetadata().getName());
        }
```